torch/nested/__init__.py

```diff
@@ -186,3 +186,34 @@ def nested_tensor(tensor_list, *, dtype=None, layout=None, device=None, requires
         return nt
     else:
         raise RuntimeError(f"Specified layout is unsupported for nested tensors: {layout}")
+
+
+
+
+def narrow(tensor: Tensor, dim: int, start: Union[SymInt, Tensor], length: Union[SymInt, Tensor], layout=torch.jagged):
```

Comment:
I don't think we should add a layout argument here. It doesn't follow the narrow syntax for regular tensors. If the user wants to change the layout they can use `to`. Since this is a view operation it shouldn't affect performance. Furthermore, layout change is likely not support when this is view only.

Reply:
> I don't think we should add a layout argument here. It doesn't follow the narrow syntax for regular tensors.

This is true, but we're also not limited to the original `narrow()` syntax since this is a new function in the `torch.nested` namespace.

So I sort of disagree. It's possible to create a view over a dense buffer with both the expanded jagged layout and the strided layout, and I think the `layout` arg is a natural extension for indicating which type of NT + metadata is to be constructed.

> If the user wants to change the layout they can use to.

Current semantics are such that this will copy, no? I don't quite see how we can guarantee `torch.nested.narrow(...).to(layout=...)` is non-copying if it's the difference between a C++ NT and a python subclass jagged layout NT.

Reply:
`torch.nested.narrow(...).to(layout=...)` can be made non-copying via torch.compile and pattern matching. 

But you're right that it's a new function. Maybe in the future we can extend torch.narrow with similar functionality.

Alright, I'm convinced.